src/string_id_null_ids.cpp

```diff
@@ -100,3 +100,4 @@ MAKE_NULL_ID2( construction, "constr_null" )
 MAKE_NULL_ID2( vehicle_prototype, "null" )
 MAKE_NULL_ID2( weakpoints, "" )
 MAKE_NULL_ID2( mutation_category_trait, "NULL" )
+MAKE_NULL_ID2( weather_generator, "" )
```

Comment:
`MAKE_NULL_ID` is for classes, `MAKE_NULL_ID2` is for structs.
```suggestion
MAKE_NULL_ID( weather_generator, "" )
```